npm_and_yarn/lib/dependabot/npm_and_yarn/file_fetcher.rb

```diff
@@ -419,11 +405,12 @@ def find_directories(glob)
       def matching_paths(glob, paths)
         ignored_glob = yarn_ignored_glob(glob)
         glob = glob.gsub(%r{^\./}, "").gsub(/!\(.*?\)/, "*")
+        glob = "#{glob}/*" if glob.end_with?("**")
 
-        results = paths.select { |filename| File.fnmatch?(glob, filename) }
+        results = paths.select { |filename| File.fnmatch?(glob, filename, File::FNM_PATHNAME) }
         return results unless ignored_glob
 
-        results.reject { |filename| File.fnmatch?(ignored_glob, filename) }
+        results.reject { |filename| File.fnmatch?(ignored_glob, filename, File::FNM_PATHNAME) }
       end
 
       def recursive_find_directories(glob, prefix = "")
```

Comment:
Is this doing what `Dir.glob` does? Is there an opportunity to simplify this now that npm_and_yarn is a cloned ecosystem?

Reply:
Yeah, this is definitely possible now indeed.

I thought of that but I wasn't clear on how to apprach tests so went with fixing (instead of changing) the current logic.

I think whenever we tackle #6923 would be a good opportunity to simplify this.